hack/kops-patch.yaml

```diff
@@ -175,3 +165,5 @@ spec:
   cloudConfig:
     awsEBSCSIDriver:
       enabled: false
+    instanceMetadata:
```

Comment:
Instead of doing this one, you can also set node affinity to run on the control plane.

Reply:
That defeats the purpose of the multi-az tests though, right?

Reply:
Only the controller needs metadata afaik. And it runs leader elected anyway.

Reply:
@olemarkus This isn't very well documented but my understanding is that currently the node service relies on the metadata service to derive the max number of attachable volumes, instance id, and AZ for dynamic provisioning.

Reply:
Both kops and EKS runs with hop limit 1 and this works... Perhaps the controller fetches this information when the daemonset cannot?

Reply:
Node can pull metadata from k8s, it only bombs out if both ec2 and k8s metadata aren't available. The reason that the controller needs metadata here is because we're using it to pull an instance IAM role.

kops can also setup IRSA instead of instance roles but requires a publicly-readable S3 bucket to do so, so it's easier to just enable IMDSv2 than jump through the hoops to get the test-infra folks to setup a bucket and modify the test scripts to use it.

Reply:
I think I'm in favor of my original solution here because it feels less janky and changing the affinity would slightly lower the test coverage (as the default affinity wouldn't be tested), but I'm not particularly committed either way and I'll let you all figure out what's best here.

Reply:
The S3 state store you are already using probably supports IRSA already. But agree IRSA is overkill.

My suggestion to run controller on the control plane is that it has a hop limit of 3, not 1. So you don't need to change the hop limit of worker nodes if you do so.